### ISSUE TITLE
🐛 Bug Report: Scaffolder template loses the form state between steps

### ISSUE THREAD

**Issue by acierto** [backstage/backstage#27316]
### 📜 Description

If to build the template where the next steps relies on the input of the previous state, it doesn't work anymore.

The example is very simple:
```

apiVersion: scaffolder.backstage.io/v1beta3
kind: Template
metadata:
  name: setup-docs
  title: Documentation (docs.stg.bol.io)
  description: Add a documentation scaffold to your project repository in GitLab registered at docs.bol.io
  tags:
    - docs
spec:
  owner: teamnorthstar
  type: service
  parameters:
    - title: Chose existing or new repository
      required:
        - existingRepository
      properties:
        existingRepository:
          type: boolean
          default: true
    - title: GitLab Settings
      required:
        - repoUrl
      properties:
        repoUrl:
          title: GitLab Repository URL
          type: string
          ui:field: GitlabCurrentOrNewProject // Here can be a default picker URL
          ui:options:
            allowedHosts:
              - gitlab.stg.bol.io
            requestUserCredentials:
              secretsKey: USER_OAUTH_TOKEN
```

So on the validation step of 'GitLab Settings' the information about the chosen repository is lost. FormData which sends the data on validator contains only 'repoUrl' field, and 'existingRepository' is not there anymore.

Though during the rendering of "GitLab Settings" those 2 properties are there.

After looking at the code I found that this change will make it work again.

<img width="1156" alt="Screenshot 2024-10-23 at 07 59 20" src="https://github.com/user-attachments/assets/bef2d668-7a35-43b9-b568-7cf824137b89">

### 👍 Expected behavior

Validator should have the context of all fields across all steps.

### 👎 Actual Behavior with Screenshots

Actual behaviour: validator has only the formData of the active step.

### 👟 Reproduction steps

Use the template provided in the example and click 'Next' -> 'Next'

The problem will be visible in the validator of the component

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by benjdlambert** [comment]
Pretty sure that this is fixed with #27273 that's in the next release, can you confirm and we can ship out a patch today?

**Comment by benjdlambert** [comment]
Closing as fixed in 1.32.3

**Comment by acierto** [comment]
Just verified, the problem is fixed
